Emphasis reading in Quickguide.

rainwoodman opened the pull request:
Attempting addressing Issue #609

tacaswell commented:
:+1: for focusing on reading rather than writing first in the quick guide (commenting on general idea, have not read actually wording closely).

andrewcollette commented on docs/quick.rst:
```diff
@@ -15,19 +15,22 @@ when using h5py is:
     **Groups work like dictionaries, and datasets work like NumPy arrays**
 
-The very first thing you'll need to do is create a new file::
+Suppose someone has send you a HDF5 file, `mytestfile.hdf5`. (If not, read createfile_ on creating a HDF5 file with h5py). The very first thing you'll need to do is to open the file::
```
typo: "sent"

andrewcollette commented on docs/quick.rst:
```diff
@@ -48,6 +51,19 @@ from a dataset in the file:
+    >>>
+    >>> f = h5py.File("mytestfile.hdf5", "w")
+
+File object has a couple of methods which look interesting. One of them is ``create_dataset``::
```
typo: "The File object"

andrewcollette commented:
Found some typos, but I think it's a good start.

andrewcollette commented:
Also just make sure to build the docs and double-check the resulting formatting is OK... this has bitten me before.

rainwoodman commented:
I am not sure if Creating a File shall be under the Core concept part or pulled to the very end of quick.rst. 

I am leaning towards pulling it the the very end of quick.rst.

andrewcollette commented:
The current version looks OK.  Do you want to add anything more or shall we merge it?

rainwoodman commented:
Thanks. I checked and fixed a few glitches. I think it's ready to go.

- Added explanation to 'create_dataset'.
- Reworded the 'context manager' sentence.

andrewcollette commented on docs/quick.rst:
```diff
@@ -48,7 +51,39 @@ from a dataset in the file:
+
+    >>> dset = f.create_dataset("mydataset", (100,), dtype='i')
+
+Now we can flush the file to the disk
```
Flushing the file is unnecessary here.  Also, "f.close()" is the canonical way to close a file, since it will succeed even if other objects are still open.

andrewcollette commented on docs/quick.rst:
```diff
@@ -48,7 +51,39 @@ from a dataset in the file:
+    >>> del f
+
+The File object is a context manager; thus we can create the file and
+manage the :code:`flush` operation with the following lines ::
```
also remove this reference to "flush".

rainwoodman replied:
What does the context manager really do other than flushing? Otherwise I will just avoid describing what it does.

andrewcollette replied:
It closes the file, equivalent to file.close().

rainwoodman replied:
By reading the code (files.py) it seems to close the file at __exit__. But what happens if I use another with block on f?

andrewcollette replied:
The file is closed at the end of the "with" block, not when the interpreter
exits.

On Fri, Aug 28, 2015 at 12:55 PM, Yu Feng <notifications@github.com> wrote:

> In docs/quick.rst
> <https://github.com/h5py/h5py/pull/610#discussion_r38232293>:
>
> >
> > +The :ref:`File object <file>` has a couple of methods which look interesting. One of them is ``create_dataset``, which
> > +as the name suggests, creates a data set of given shape and dtype ::
> > +
> > +    >>> dset = f.create_dataset("mydataset", (100,), dtype='i')
> > +
> > +Now we can flush the file to the disk
> > +
> > +    >>> f.flush()
> > +    >>> del f
> > +
> > +The File object is a context manager; thus we can create the file and
> > +manage the :code:`flush` operation with the following lines ::
>
> By reading the code (files.py) it seems to close the file at *exit*. But
> what happens if I use another with block on f?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/h5py/h5py/pull/610/files#r38232293>.
>

andrewcollette commented:
A few other corrections and I think we're good to go.

andrewcollette commented:
Thanks again for contributing!

rainwoodman commented:
Thanks. It's been a pleasure working with you.
